.github/workflows/deploy.yml

```diff
@@ -0,0 +1,163 @@
+name: Deploy
+
+on:
+  push:
+    branches:
+      - master
+
+  # Run actions on PRs, but only deploy on master
+  pull_request:
+
+jobs:
+  cargo_docs:
+    name: Cargo Docs
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - uses: actions-rs/toolchain@v1
+        id: toolchain
+        with:
+          toolchain: stable
+          profile: minimal
+          components: rust-docs
+          override: true
+
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: doc-${{ runner.os }}-${{ steps.toolchain.outputs.rustc_hash }}-${{ hashFiles('**/Cargo.lock') }}
+
+      - run: make doc
+
+      - name: Deploy
+        if: github.ref == 'refs/heads/master'
+        uses: peaceiris/actions-gh-pages@v3
```

Comment:
this has come up before but do we have a general policy wrt third-party github actions? since github tags are mutable as opposed to most package registries, and this has access to a token

Reply:
Deferring to the policy-makers here. So far, there has been no push-back. 

Note that this was not changed from master, just moved.

Reply:
Maybe @billyvg has insights?